### PR TITLE
Add audit logging for UserAccountCreationController

### LINKED ISSUE #1925: AccountRequestController and UserAccountCreationController do not create AuditLog entries
Our Spring REST controllers should produce audit logs. The PatientExperienceController does so using this annotation:

```
@PostAuthorize("@restAuditLogManager.logRestSuccess(#request, returnObject)")
```

AccountRequestController and UserAccountCreationController are not so annotated, and do not appear to have explicit audit logging built in. These controllers are used mostly for anonymous requests, so one might argue that they don't need to appear in the log... but we should probably cover our bases

### PR DESCRIPTION
**Pull request by emmastephenson**
## Related Issue or Background Info

Fixes #1925 

## Changes Proposed

Adds audit logging to AccountRequestController and UserAccountCreationController.

## Additional Information

- these requests are effectively anonymous (because users aren't signed in.) I created a new anonymous ApiUser to handle this case, because it's required in the audit log.
- if possible, I do log the Okta user id (this is applicable to most but not all UserAccountCreation requests).
- We don't ever log the request body with these requests, so we shouldn't have any issues around logging passwords or other sensitive information.

### Testing
- [x] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- [x] Database changes are submitted as a separate PR
  - [x] Any new tables that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
  - [x] Any changes to tables that have custom no-PHI views are accompanied by changes to those views
        (including re-granting permission to the no-PHI user if need be)
  - [x] Liquibase rollback has been tested locally using `./gradlew liquibaseRollbackSQL` or `liquibaseRollback`
- [x] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [x] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [x] Any dependencies introduced have been vetted and discussed

## Cloud
- [ ] DevOps team has been notified if PR requires ops support
- [x] If there are changes that cannot be tested locally, this has been deployed to our Azure `test` environment for verification

### CODE REVIEW

**Review by jdorothy — APPROVED** [review]
LGTM!

File: backend/src/main/java/gov/cdc/usds/simplereport/service/ApiUserService.java [hunk 1]
```diff
@@ -313,6 +314,21 @@ private ApiUser getAccountRequestApiUser() {
         });
   }
 
+  /** Only used for audit logging. */
+  public ApiUser getAnonymousApiUser() {
+    Optional<ApiUser> found = _apiUserRepo.findByLoginEmail(ANONYMOUS_EMAIL);
+    return found.orElseGet(
+        () -> {
+          ApiUser magicUser =
```
Comment (jdorothy) [inline backend/src/main/java/gov/cdc/usds/simplereport/service/ApiUserService.java:322]: What is the magic user exactly? Is this just a placeholder for logging anonymous requests?

Comment (emmastephenson) [reply backend/src/main/java/gov/cdc/usds/simplereport/service/ApiUserService.java:322]: Yep, exactly. The audit log has `api user` as a required field in the database, so we need a pseudo-user to record anonymous requests.

Comment (jdorothy) [reply backend/src/main/java/gov/cdc/usds/simplereport/service/ApiUserService.java:322]: Makes sense - thanks!

**Pull request merged**